src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -159,6 +161,24 @@
     }
   }
 
+  public List<String> getRefIdsForSnapshot(Snapshot snapshot) {
+    List<String> refIds = new ArrayList<>();
+    snapshot
+        .getTables()
+        .forEach(
+            table -> {
+              if (table.getRowCount() > 0) {
+                String tableName =
+                    IngestUtils.formatSnapshotTableName(snapshot.getId(), table.getName());
+                table.getColumns().stream()
+                    .map(Column::toSynapseColumn)
+                    .filter(Column::isFileOrDirRef)
+                    .forEach(column -> refIds.addAll(getRefIds(tableName, column)));
+              }
+            });
+    return refIds;
```

Comment:
You can convert the side-effecting loop into a stream-only operation by using `flatMap()`. Up to you if you think it's better or not.
```suggestion
    return snapshot.getTables().stream()
        .filter(table -> table.getRowCount() > 0)
        .flatMap(
            table -> {
              String tableName =
                  IngestUtils.formatSnapshotTableName(snapshot.getId(), table.getName());
              return table.getColumns().stream()
                  .map(Column::toSynapseColumn)
                  .filter(Column::isFileOrDirRef)
                  .flatMap(column -> getRefIds(tableName, column).stream());
            }).collect(Collectors.toList());
```

Reply:
That's much cleaner, thanks!